addons/mail/models/mail_channel.py

```diff
@@ -920,7 +930,10 @@ def channel_get(self, partners_to, pin=True):
             channel = self.browse(result[0].get('channel_id'))
             # pin up the channel for the current partner
             if pin:
-                self.env['mail.channel.partner'].search([('partner_id', '=', self.env.user.partner_id.id), ('channel_id', '=', channel.id)]).write({'is_pinned': True})
+                self.env['mail.channel.partner'].search([('partner_id', '=', self.env.user.partner_id.id), ('channel_id', '=', channel.id)]).write({
+                    'is_pinned': True,
+                    'last_activity_time': fields.Datetime.now(),
```

Comment:
If this update is systematic to update of is_pinned maybe it should be done in either an override of write, or in a specific method to pin and update this value?